src/Composer/Satis/PackageSelection/PackageSelection.php

```diff
@@ -137,6 +171,10 @@ public function select(Composer $composer, $verbose)
             }
         }
 
+        if($this->hasRepositoryFilter() && count($repos) !== 1) {
+            throw new \InvalidArgumentException(sprintf('Specified repository url %s does not exist.', $this->repositoryFilter));
```

Comment:
Please use quotes (`... url "%s" does ...`) to clearly show where the URL starts and ends.